reactor-netty-core/src/main/java/reactor/netty/resources/LoopResources.java

```diff
@@ -175,6 +175,18 @@ static LoopResources create(String prefix, int selectCount, int workerCount, boo
 		return new DefaultLoopResources(prefix, selectCount, workerCount, daemon, colocate);
 	}
 
+	/**
+	 * Return a {@link LoopResources} that will share the same {@link EventLoopGroup} from this {@link LoopResources}, but with
+	 * colocation support disabled (if it was enabled).<p>
+	 * Disposing the returned {@link LoopResources} will also dispose the {@link EventLoopGroup} from the origin {@link LoopResources}.
+	 *
+	 * @return a LoopResources that will share the same {@link EventLoopGroup} from this {@link LoopResources}
+	 * @since 1.1.9
+	 */
+	default LoopResources disableColocation() {
```

Comment:
@pderop This interface `LoopResources` can have different implementations. The proposed change `LoopResources.disableColocation()` is specific to the implementation in Reactor Netty only (`DefaultLoopResources`, `TcpResources` etc). In other implementations (for example https://github.com/turms-im/turms/blob/b2a817058b45b5e8bbb8c0a6e5808e92bfbbb884/turms-server-common/src/main/java/im/turms/server/common/access/common/LoopResourcesFactory.java#L30) this may not be the case and colocation might not exist as a configuration at all.
I suggest that we provide a change that can fit also with other implementations if the change is on the interface level.

Reply:
What do you think if I  do not do the API change at the interface (instance) level, and instead of that, since we already have some `create(...)` methods which are using the ReactorNetty specific implementations (DefaultLoopResources, TcpResources, ...), then just add a new static create method like this in the LoopResources:

```
	/**
	 * Return a {@link LoopResources} that will share the same {@link EventLoopGroup} from an existing parent {@link LoopResources}.
	 * Disposing the returned {@link LoopResources} will also dispose the {@link EventLoopGroup} from the parent as well.
	 *
	 * @param parent the parent {@link LoopResources} whose underlying {@link EventLoopGroup} will be reused by the returned {@link LoopResources}
	 * @param colocate <code>false</code> will disable colocation from the parent {@link EventLoopGroup} if it was enabled in the parent.
	 *                 <code>true</code> means colocation will be enabled if it was disabled in the parent. The parent is not affected whether <code>true</code>
	 *                 or <code>false</code> is used.
	 * @return a LoopResources that will share the same {@link EventLoopGroup} from the specified parent, with or without colocation mode
	 * @since 1.1.9
	 */
	static LoopResources create(LoopResources parent, boolean colocate) {
		return new DefaultLoopResources(parent, colocate);
	}

```

This new static method is consistent with the other `create` methods that are depending on our ReactorNetty specfic implementations, and also, now we have a new
`DefaultLoopResources(LoopResources parent, boolean colocate)` constructor that is clearer than having a single constructor parameter `DefaultLoopResources(LoopResources parent)` that was implicitly disabling colocation.

would this be OK, and am I missing your point ?

Reply:
If we have in the future a request for changing other configurations how the API will look like?

Reply:
You can ignore my last comment, indeed doing so will lead to a kind of bloated API, because in the future, if we have more new re-configurable configs, then we would end up
with many create method signatures like this, which would ruin the original cohesive feel of the API:

```
create(parent, boolean /* colocate */)
create(parent, boolean /* colocate */, boolean /* enable/disable new configuration1 */)
create(parent, boolean /* colocate */, boolean /* enable/disable new configuration1 */, boolean /* enable/disable new configuration2 */)
etc ...
```

in this case, a Builder API would be more appropriate.

But, I got back to your initial suggestion from https://github.com/reactor/reactor-netty/pull/2842#discussion_r1241802262:

> I suggest that we provide a change that can fit also with other implementations if the change is on the interface level.

and in the last commit, I revisited things, and it might be more aligned with what you have in mind, can you recheck ?

- first, I have reverted all changes from DefaultLoopResources, because in fact, it is not necessary to add complexity to this class.
- the LoopResources.disableColocation method is now using a helper class which is independant of our current Reactor-Netty LoopResources implementations (like DefaultLoopResources, TcpResources, ...)
- it should fit to other implementations like the one from `turms`; 
for example, if the turms `LoopResourcesFactory` has support for colocation and implements a `onClient(boolean useNative)` method that returns an EventLoopGroup decorated using `LoopResources.colocate(EventLoopGroup group)`, then the disableColocation() method shoud fit ?

let me know ?

Reply:
ok, so I'll continue to think about this.